### PR TITLE
Add SweetCollectionView and related classes

### PR DESCRIPTION
**Pull request by elland**
This aims to be a simple drop-in replacement for the provided `UICollectionViewController`, with the improvements of:
- having a non-optional collection view
- the collection view is not the backing view, so it's easier to extend the layout in code without resorting to view containment like with storyboards
- the ability to easily set it up with a custom collection view layout

Hopefully with more to come as we find out more cases we can generalise and improve.

### CODE REVIEW

**Comment by 3lvis** [comment]
https://github.com/bakkenbaeck/SweetUIKit/pull/9 :)

**Comment by elland** [comment]
🎉

File: Demo/ViewController.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import UIKit
+
+class CollectionCell: UICollectionViewCell {
```
Comment (marijnschilling) [inline Demo/ViewController.swift]: Would be great if you'd put this cell in a different file, I think that is part of our best practices right?

Comment (elland) [reply Demo/ViewController.swift]: Yeah, I was thinking about it. Despite being just a very simplified demo. I want to make this demo better, and I thought: mhm, that could be a great place to do the Unsplash thing! What do you think? cc @3lvis

Comment (3lvis) [reply Demo/ViewController.swift]: Demoing this in Unsplash sounds like a great idea! But we still need to have it here so we can run some unit test or UI tests.

Comment (elland) [reply Demo/ViewController.swift]: I was thinking of having the Unsplash thing _be_ the SweetUIKit demo app. In here.

Comment (3lvis) [reply Demo/ViewController.swift]: I think it's probably better if we have Unsplash in a separate repo, too many dependencies. It's probably better if here we have a very lightweight demo.

Comment (elland) [reply Demo/ViewController.swift]: 👌

**Comment by elland** [comment]
What do you think of this then? Good enough? Should it cover more? Ideas for how to improve this demo?

File: Demo/AppDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        let window = UIWindow()
+        window.rootViewController = ViewController()
+        window.makeKeyAndVisible()
+        self.window = window
+        
+        return true
+    }
+
+    func applicationWillResignActive(application: UIApplication) {
+        // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
```
Comment (3lvis) [inline Demo/AppDelegate.swift]: Please delete all the boilerplate here

Comment (3lvis) [inline Demo/AppDelegate.swift]: Bye empty methods!

File: Demo/ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+class ViewController: SweetCollectionViewController {
+
+    lazy var data: [String] = {
+        return ["123", "321", "lol"]
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view, typically from a nib.
```
Comment (3lvis) [inline Demo/ViewController.swift]: Could you remove this comment?

File: Demo/ViewController.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import UIKit
+
+class ViewController: SweetCollectionViewController {
+
```
Comment (3lvis) [inline Demo/ViewController.swift]: Could you remove this newline, please?

File: Demo/ViewController.swift [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    lazy var data: [String] = {
+        return ["123", "321", "lol"]
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view, typically from a nib.
+        self.view.backgroundColor = .magentaColor()
+
+        self.collectionView.register(CollectionCell.self)
```
Comment (3lvis) [inline Demo/ViewController.swift:13]: This is so beautiful

Comment (elland) [reply Demo/ViewController.swift:13]: 😍

File: Demo/ViewController.swift [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        self.collectionView.dataSource = self
+    }
+}
+
+extension ViewController: UICollectionViewDataSource {
+    func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCellWithReuseIdentifier(CollectionCell.reuseIdentifier, forIndexPath: indexPath) as! CollectionCell
+        cell.backgroundColor = .whiteColor()
+        cell.label.text = self.data[indexPath.row]
+        return cell
```
Comment (3lvis) [inline Demo/ViewController.swift:25]: Please add a newline before the return 👮

File: Demo/ViewController.swift [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return self.data.count
+    }
+}
+
+extension ViewController: UICollectionViewDelegate {
```
Comment (3lvis) [inline Demo/ViewController.swift]: I think it's safe to drop the empty delegate implementation 💭

**Comment by 3lvis** [comment]
Besides the pointed comments I think this seems a pretty solid start for something great.

File: Demo/ViewController.swift (hunk 7 above)
Comment (elland) [reply Demo/ViewController.swift]: We're keeping the demo minimal here?

**Comment by 3lvis** [comment]
giphy-1

File: Demo/ViewController.swift (hunk 7 above)
Comment (3lvis) [reply Demo/ViewController.swift]: why not? :D

File: Demo/CollectionCell.swift [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    lazy var label: UILabel = {
+        let view = UILabel(withAutoLayout: true)
+        view.textAlignment = .Center
+
+        return view
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: .zero)
+        self.contentView.addSubview(self.label)
```
Comment (3lvis) [inline Demo/CollectionCell.swift:14]: Newline after super.init, please

File: Demo/ViewController.swift [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import UIKit
+
+class ViewController: SweetCollectionViewController {
+    lazy var data: [String] = {
+        return ["123", "321", "lol"]
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.view.backgroundColor = .magentaColor()
```
Comment (3lvis) [inline Demo/ViewController.swift:11]: Could you add a newline after super.viewDidLoad :D ?

**Comment by 3lvis** [comment]
Bravo!

**Comment by 3lvis** [comment]
💥 https://github.com/bakkenbaeck/SweetUIKit/releases/tag/0.3.0

**Pull request merged**